File: GFG/POTD_Alternatingsort25October.py
class Solution:
    def alternateSort(self,arr):
        arr.sort()
        n=len(arr)
        a,b=0,0
        largest_number=list()
        smallest_number=list()
        result=list()
        mid_point=n/2  
        for index in range(len(arr)):
            if index<mid_point:
                smallest_number.append(arr[index])
            else:
                largest_number.append(arr[index])
        largest_number.reverse()
        for i in range(len(arr)):
            if i % 2 == 0 and a < len(largest_number):
                result.append(largest_number[a])
                a=a+1
            elif b < len(smallest_number):  
                result.append(smallest_number[b])
                b=b+1
                
        return result

File: GFG/POTD_Alternatingsort25October.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

class AlternatingSort{
    public:
        
        vector<int> alternateSort(vector<int>& arr){
        vector<int>result;
        int mid_point;
        int n=arr.size();
        int a=0;
        int b=0;
        vector<int>largest_numbers;
        vector<int>smallest_numbers;
        sort(arr.begin(), arr.end());
        mid_point = n / 2;
        for(int i=0;i<n;i++){
            if(i<mid_point){
                smallest_numbers.push_back(arr[i]);
                }
            else{
                largest_numbers.push_back(arr[i]);
                }
            }
        reverse(largest_numbers.begin(), largest_numbers.end());
        for(int i=0; i<smallest_numbers.size()+largest_numbers.size(); i++){
            if(i%2==0){
                    result.push_back(largest_numbers[a]);
                    a++;
                }
            else{
                    result.push_back(smallest_numbers[b]);
                    b++;
                }
            }
            return result;
        }
};


int main(){
    AlternatingSort obj;
    vector<int> a={7, 1, 2, 3, 4, 5, 6};
    vector<int> res=obj.alternateSort(a);
    for(int x: res){
        cout<<x<<endl;
    }
}